### PR TITLE
Hair - Exposing Marschner lighting control parameters and connecing U…

### PR DESCRIPTION
**Pull request by Adi-Amazon**
…I to shader

### CODE REVIEW

**Review by amzn-rhhong — APPROVED** [review]

File: Gems/AtomTressFX/Assets/Shaders/HairRenderingSrgs.azsli [hunk 1]
```diff
@@ -63,15 +63,19 @@ struct TressFXRenderParameters
...
-    float3      m_fPadding0;
-             
+    float3      m_fPadding0;           
     float4      m_matKValue;
 
     int         m_maxShadowFibers;
-    int3        m_iPadding0;
+
+    // Marschner lighting model parameters 
+    float		m_roughness;
```
Comment (Adi-Amazon) [inline Gems/AtomTressFX/Assets/Shaders/HairRenderingSrgs.azsli:75]: turned out copy paste of white spaces kept tabs - fixed

File: Gems/AtomTressFX/Assets/Shaders/HairRenderingSrgs.azsli [hunk 2]
```diff
@@ -170,13 +174,17 @@ struct HairObjectShadeParams
...
-    // For lighting/shading
+
+    // Original TressFX Kajiya lighting model parameters
     float       m_hairEx2;
     float4		m_matKValue;   // KAmbient, KDiffuse, KSpec1, Exp1
     float       m_hairKs2;
-    float		fPadding0;
+
+    // Marschner lighting model parameters 
     float		m_roughness;
```
Comment (Adi-Amazon) [inline Gems/AtomTressFX/Assets/Shaders/HairRenderingSrgs.azsli:184]: same here - changed to spaces

File: Gems/AtomTressFX/External/Code/src/TressFX/TressFXConstantBuffers.h [hunk 3]
```diff
@@ -145,18 +145,23 @@ namespace AMD
...
         float       HairEx2 = 11.80f;
         float3      fPadding0;
 
         float4      MatKValue = { 0.f, 0.07f, 0.0017f, 14.40f };   // KAmbient, KDiffuse, KSpec1, Exp1
 
         int         MaxShadowFibers = 50;
-        sint3       iPadding0;
+
+        // Marschner lighting model parameters 
+        float		Roughness;
```
Comment (Adi-Amazon) [inline Gems/AtomTressFX/External/Code/src/TressFX/TressFXConstantBuffers.h:158]: and here - white spaces

File: Gems/AtomTressFX/External/Code/src/TressFX/TressFXConstantBuffers.h [hunk 4]
```diff
@@ -196,13 +201,17 @@ namespace AMD
...
+
+        // Original TressFX Kajiya lighting model parameters
         float       HairEx2 = 11.80f;
         float4      MatKValue = { 0.f, 0.07f, 0.0017f, 14.40f };   // KAmbient, KDiffuse, KSpec1, Exp1
         float       HairKs2 = 0.072f;
+
+        // Marschner lighting model parameters 
+        float		Roughness;
+        float		CuticleTilt;  // tile angle in radians
+
```
Comment (Adi-Amazon) [inline Gems/AtomTressFX/External/Code/src/TressFX/TressFXConstantBuffers.h:213]: same

**Pull request merged**